### PR TITLE
OHRM5X-1846: Develop buzz - likes on comment API

### PR DESCRIPTION
**Pull request by devishke-orange**
PR contains:
- OHRM5X-1846: [BE][API] Comment Likes API::getAll
- OHRM5X-1847: [BE][API] Comment Likes API::create
- OHRM5X-1848: [BE][API] Comment Likes API::update

### CODE REVIEW

**Review by ManulMax — APPROVED** [review]

File: src/plugins/orangehrmBuzzPlugin/Dao/BuzzLikeDao.php [hunk 1]
```diff
@@ -72,42 +101,98 @@ public function getBuzzLikeOnShareByShareIdAndEmpNumber(int $shareId, int $empNu
     }
 
     /**
-     * @param BuzzLikeSearchFilterParams $buzzLikeSearchFilterParams
+     * @param int $commentId
+     * @param int $empNumber
+     * @return BuzzLikeOnComment|null
+     */
+    public function getBuzzLikeOnCommentByShareIdAndEmpNumber(int $commentId, int $empNumber): ?BuzzLikeOnComment
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Dao/BuzzLikeDao.php:108]: *

File: src/plugins/orangehrmBuzzPlugin/Dto/BuzzLikeOnCommentSearchFilterParams.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Boston, MA 02110-1301, USA
+ */
+
+namespace OrangeHRM\Buzz\Dto;
+
+use OrangeHRM\Core\Dto\FilterParams;
+
+class BuzzLikeOnCommentSearchFilterParams extends FilterParams
+{
+    public const ALLOWED_SORT_FIELDS = ['commentLike.id'];
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Dto/BuzzLikeOnCommentSearchFilterParams.php:26]: *

File: src/plugins/orangehrmBuzzPlugin/Dto/BuzzLikeOnShareSearchFilterParams.php [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 use OrangeHRM\Core\Dto\FilterParams;
 
-class BuzzLikeSearchFilterParams extends FilterParams
+class BuzzLikeOnShareSearchFilterParams extends FilterParams
 {
     public const ALLOWED_SORT_FIELDS = ['shareLike.id'];
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Dto/BuzzLikeOnShareSearchFilterParams.php:26]: *

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnCommentAPI.php [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+     *             ),
+     *             @OA\Property(
+     *                 property="meta",
+     *                 type="object",
+     *                 @OA\Property(property="total", type="integer")
+     *             )
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response="422",
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnCommentAPI.php:89]: *

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnCommentAPI.php [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+            $buzzComment = $this->getBuzzService()->getBuzzDao()->getBuzzCommentById($commentId);
+            if (!$buzzComment instanceof BuzzComment) {
+                throw $this->getInvalidParamException(self::PARAMETER_COMMENT_ID);
+            }
+
+            $buzzShareOnComment = $this->getBuzzService()
+                ->getBuzzLikeDao()
+                ->getBuzzLikeOnCommentByShareIdAndEmpNumber($commentId, $this->getAuthUser()->getEmpNumber());
+            if ($buzzShareOnComment instanceof BuzzLikeOnComment) {
+                throw $this->getBadRequestException('Comment is already liked');
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnCommentAPI.php:229]: *

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnCommentAPI.php [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+            $buzzComment = $this->getBuzzService()->getBuzzDao()->getBuzzCommentById($commentId);
+            if (!$buzzComment instanceof BuzzComment) {
+                throw $this->getInvalidParamException(self::PARAMETER_COMMENT_ID);
+            }
+
+            $buzzCommentOnLike = $this->getBuzzService()
+                ->getBuzzLikeDao()
+                ->getBuzzLikeOnCommentByShareIdAndEmpNumber($commentId, $this->getAuthUser()->getEmpNumber());
+            if (!$buzzCommentOnLike instanceof BuzzLikeOnComment) {
+                throw $this->getBadRequestException('Comment is not liked');
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnCommentAPI.php:359]: *

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnCommentAPI.php [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+            $this->getBuzzService()->getBuzzLikeDao()->deleteBuzzLikeOnComment($commentId, $this->getAuthUser()->getEmpNumber());
+            $this->commitTransaction();
+
+            return new EndpointResourceResult(ArrayModel::class, [self::PARAMETER_COMMENT_ID => $commentId]);
+        } catch (InvalidParamException $invalidParamException) {
+            $this->rollBackTransaction();
+            throw $invalidParamException;
+        } catch (BadRequestException $badRequestException) {
+            $this->rollBackTransaction();
+            throw $badRequestException;
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnCommentAPI.php:374]: *

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnShareAPI.php [hunk 8]
```diff
@@ -205,31 +211,45 @@ public function getValidationRuleForGetAll(): ParamRuleCollection
...
 
-        $like = $this->getBuzzService()->getBuzzLikeDao()->saveBuzzLikeOnShare($like);
-        return new EndpointResourceResult(BuzzLikeOnShareModel::class, $like);
+            return new EndpointResourceResult(BuzzLikeOnShareModel::class, $like);
+        } catch (InvalidParamException $invalidParamException) {
+            $this->rollBackTransaction();
+            throw $invalidParamException;
+        } catch (BadRequestException $badRequestException) {
+            $this->rollBackTransaction();
+            throw $badRequestException;
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnShareAPI.php:248]: *

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**